core/embed/rust/src/ui/model_mercury/component/number_input_slider.rs

```diff
@@ -76,24 +76,27 @@ impl Component for NumberInputSliderDialog {
     }
 
     fn event(&mut self, ctx: &mut EventCtx, event: Event) -> Option<Self::Msg> {
-        if let Some(value) = self.input.event(ctx, event) {
-            self.val = value;
+        let msg_opt = self.input.event(ctx, event);
+
+        if self.val == self.init_val || self.input.touching {
+            self.footer
+                .update_instruction(ctx, TR::instructions__swipe_horizontally);
+            self.footer.update_description(ctx, TR::setting__adjust);
+            ctx.request_paint();
+        } else {
+            self.footer
+                .update_instruction(ctx, TR::instructions__swipe_up);
+            self.footer.update_description(ctx, TR::setting__apply);
+            ctx.request_paint();
+        }
 
-            if self.val == self.init_val || self.input.touching {
-                self.footer
-                    .update_instruction(ctx, TR::instructions__swipe_horizontally);
-                self.footer.update_description(ctx, TR::setting__adjust);
-                ctx.request_paint();
-            } else {
-                self.footer
-                    .update_instruction(ctx, TR::instructions__swipe_up);
-                self.footer.update_description(ctx, TR::setting__apply);
-                ctx.request_paint();
+        match msg_opt {
```

Comment:
Nit: if you wanted to use less lines:
```rust
        msg_opt.map(|value| {
            self.val = value;
            Self::Msg::Changed(value)
        })
```
But I'm personally fine with the current state.

Reply:
[921e065](https://github.com/trezor/trezor-firmware/pull/4279/commits/921e06552aae9ff0854a49e53425fda65f023f80)